### ISSUE TITLE
Jackson support and nulls 

### ISSUE THREAD

**Issue by xtaixe** [immutables/immutables#1309]
Hi.

Is there any way to allow nulls (specifically a map value in our case) when deserializing JSON with Jackson? Even with `@Value.Style(validationMethod = NONE, jdkOnly = true)` there are still a bunch of `requireNonNull`. Is that expected? Is this something that can be fixed or is there any workaround?

Not being able to read the whole object and validate later is a bit of a problem for us...

Thanks.

**Comment by elucash** [comment]
Can you specify places where there is null checks. With validation set to none we don't do null checks in general, but maybe we've overlooked some places, it also can be caused by encoding extensions (i don't know if you use any).

**Comment by xtaixe** [comment]
If by encoding extensions you mean what's documented here: https://immutables.github.io/encoding.html, we don't use that.

I have the following:

```java
@Value.Immutable
@Value.Style(validationMethod = NONE, jdkOnly = true)
@JsonSerialize(as = ImmutableTestContainers.class)
@JsonDeserialize(as = ImmutableTestContainers.class)
public interface TestContainers {
    Map<String, String> getMap();

    List<String> getList();

    Optional<String> getOptional();
}
```

And I get the following in `ImmutableTestContainers `:

```java
public final Builder putMap(String key, String value) {
    this.map.put(
        Objects.requireNonNull(key, "map key"),
        Objects.requireNonNull(value, "map value"));
    return this;
}

public final Builder putMap(Map.Entry<String, ? extends String> entry) {
    String k = entry.getKey();
    String v = entry.getValue();
    this.map.put(
        Objects.requireNonNull(k, "map key"),
        Objects.requireNonNull(v, "map value"));
    return this;
}

public final Builder putAllMap(Map<String, ? extends String> entries) {
    for (Map.Entry<String, ? extends String> e : entries.entrySet()) {
      String k = e.getKey();
      String v = e.getValue();
      this.map.put(
          Objects.requireNonNull(k, "map key"),
          Objects.requireNonNull(v, "map value"));
    }
    return this;
}

// Calls createUnmodifiableMap with checkNulls==true
public final ImmutableTestContainers withMap(Map<String, ? extends String> entries) {
  if (this.map == entries) return this;
  Map<String, String> newValue = createUnmodifiableMap(true, false, entries);
  return new ImmutableTestContainers(newValue, this.list, this.optional);
}
```

**Comment by xtaixe** [comment]
@elucash any further thoughts or guidance on what is going to happen with this? Thanks.

**Comment by elucash** [comment]
sorry for the delay, I think this is bug/shortcoming to be fixed: null-checks should not be inserted for collections (jdk only, as guava's immutable collections will fail on null anyway) if `validationMethod = NONE`

**Comment by elucash** [comment]
apparently maps was not handled properly (so far), the fix released in `2.9.0-beta3`

**Comment by xtaixe** [comment]
Cool. Thanks. Any estimates on when will `2.9.0` stable be released? Thanks.

**Comment by elucash** [comment]
my estimate is in 1–1.5 month (judging from the current trajectory), but can be earlier. I can just say that that current beta is quite stable compared to what will be released, we're adding only "safe" fixes or small/safe "bug-driven features".

**Comment by elucash** [comment]
took a lot more, apparently, released in 2.9.0
